File: Shiny Nemesis/src/ch/bulletproof/countdown/announcer/Announcement.java
package ch.bulletproof.countdown.announcer;


/**
 * @author tourn
 * Represents a single announcement made by an announcer
 */
public abstract class Announcement {
	/**
	 * The amount of seconds this Announcement happens before the end of the countdown
	 */
	protected int secondsToEnd;
	
	/**
	 * Executes the Media action of this Announcement: Toasts, Vibration, Sounds...
	 */
	public abstract void play();

	public int getSecondsToEnd() {
		return secondsToEnd;
	}

	public void setSecondsToEnd(int secondsToEnd) {
		this.secondsToEnd = secondsToEnd;
	}
	
}

File: app/src/main/java/ch/trq/countdown/announcer/Announcement.kt
package ch.trq.countdown.announcer

/**
 * @author tourn
 * Represents a single announcement made by an announcer
 */
abstract class Announcement {
    /**
     * The amount of seconds this Announcement happens before the end of the countdown
     */
    var secondsToEnd = 0

    /**
     * Executes the Media action of this Announcement: Toasts, Vibration, Sounds...
     */
    abstract fun play()
}